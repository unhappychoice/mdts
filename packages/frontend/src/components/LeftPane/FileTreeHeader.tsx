import React from 'react';
import { Box, IconButton, Typography } from '@mui/material';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import UnfoldLessIcon from '@mui/icons-material/UnfoldLess';
import UnfoldMoreIcon from '@mui/icons-material/UnfoldMore';

interface FileTreeHeaderProps {
  isOpen: boolean;
  onToggle: () => void;
  onExpandAllClick: () => void;
  onCollapseAll: () => void;
}

const FileTreeHeader: React.FC<FileTreeHeaderProps> = ({ isOpen, onToggle, onExpandAllClick, onCollapseAll }) => {
  return (
    <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', height: '34px', marginTop: isOpen ? '0' : '16px', marginBottom: 2 }}>
      {isOpen && (
        <Box sx={{ display: 'flex', alignItems: 'center', flex: 1 }}>
          <Typography variant="h6" gutterBottom sx={{ flex: 1, marginLeft: 1, marginBottom: 0 }}>
            File Tree
          </Typography>
          <IconButton onClick={onExpandAllClick} size="small" aria-label="expand all">
            <UnfoldMoreIcon />
          </IconButton>
          <IconButton onClick={onCollapseAll} size="small" aria-label="collapse all">
            <UnfoldLessIcon />
          </IconButton>
        </Box>
      )}
      <IconButton onClick={onToggle} size="small" sx={{ marginBottom: 0, marginLeft: isOpen ? '0' : '12px' }} aria-label="toggle file tree">
        {isOpen ? <ChevronLeftIcon /> : <ChevronRightIcon />}
      </IconButton>
    </Box>
  );
};

export default FileTreeHeader;
