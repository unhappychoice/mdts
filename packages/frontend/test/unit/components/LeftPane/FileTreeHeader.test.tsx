import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import FileTreeHeader from '../../../../src/components/LeftPane/FileTreeHeader';

describe('FileTreeHeader', () => {
  test('renders correctly when open', () => {
    const { asFragment } = render(
      <FileTreeHeader
        isOpen={true}
        onToggle={jest.fn()}
        onExpandAllClick={jest.fn()}
        onCollapseAll={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('File Tree')).toBeInTheDocument();
    expect(screen.getByLabelText('expand all')).toBeInTheDocument();
    expect(screen.getByLabelText('collapse all')).toBeInTheDocument();
  });

  test('renders correctly when closed', () => {
    const { asFragment } = render(
      <FileTreeHeader
        isOpen={false}
        onToggle={jest.fn()}
        onExpandAllClick={jest.fn()}
        onCollapseAll={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.queryByText('File Tree')).not.toBeInTheDocument();
    expect(screen.queryByLabelText('expand all')).not.toBeInTheDocument();
    expect(screen.queryByLabelText('collapse all')).not.toBeInTheDocument();
  });

  test('calls onToggle when toggle button is clicked', () => {
    const onToggleMock = jest.fn();
    render(
      <FileTreeHeader
        isOpen={true}
        onToggle={onToggleMock}
        onExpandAllClick={jest.fn()}
        onCollapseAll={jest.fn()}
      />
    );
    fireEvent.click(screen.getByTestId('ChevronLeftIcon'));
    expect(onToggleMock).toHaveBeenCalled();
  });

  test('calls onExpandAllClick when expand all button is clicked', () => {
    const onExpandAllClickMock = jest.fn();
    render(
      <FileTreeHeader
        isOpen={true}
        onToggle={jest.fn()}
        onExpandAllClick={onExpandAllClickMock}
        onCollapseAll={jest.fn()}
      />
    );
    fireEvent.click(screen.getByLabelText('expand all'));
    expect(onExpandAllClickMock).toHaveBeenCalled();
  });

  test('calls onCollapseAll when collapse all button is clicked', () => {
    const onCollapseAllMock = jest.fn();
    render(
      <FileTreeHeader
        isOpen={true}
        onToggle={jest.fn()}
        onExpandAllClick={jest.fn()}
        onCollapseAll={onCollapseAllMock}
      />
    );
    fireEvent.click(screen.getByLabelText('collapse all'));
    expect(onCollapseAllMock).toHaveBeenCalled();
  });
});
