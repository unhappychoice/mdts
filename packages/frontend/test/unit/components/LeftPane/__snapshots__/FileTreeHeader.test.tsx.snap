// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FileTreeHeader renders correctly when closed 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-luhtu9"
  >
    <button
      aria-label="toggle file tree"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-130mf42-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="ChevronRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`FileTreeHeader renders correctly when open 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-xonpca"
  >
    <div
      class="MuiBox-root css-1g3qtqm"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1b6r7r0-MuiTypography-root"
      >
        File Tree
      </h6>
      <button
        aria-label="expand all"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="UnfoldMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15z"
          />
        </svg>
      </button>
      <button
        aria-label="collapse all"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="UnfoldLessIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 18.59 8.83 20 12 16.83 15.17 20l1.41-1.41L12 14zm9.18-13.18L15.17 4 12 7.17 8.83 4 7.41 5.41 12 10z"
          />
        </svg>
      </button>
    </div>
    <button
      aria-label="toggle file tree"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1nyqon6-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="ChevronLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
